plugins/transformer/basic/client/schemaupdater/schema_updater.go

```diff
@@ -95,3 +96,18 @@ func (s *SchemaUpdater) ChangeTableName(newTableNamePattern string) (*arrow.Sche
 	newMetadata := arrow.MetadataFrom(m)
 	return arrow.NewSchema(s.schema.Fields(), &newMetadata), nil
 }
+
+func (s *SchemaUpdater) AddPrimaryKeys(newPks []string) (*arrow.Schema, error) {
+	table, err := schema.NewTableFromArrowSchema(s.schema)
+	if err != nil {
+		return nil, err
+	}
+	for _, newPk := range newPks {
+		newCol := table.Columns.Get(newPk)
+		if newCol == nil {
+			return nil, errors.New("new primary key column not found in schema")
```

Comment:
Can we add the missing column name to the error so it's easier to know what caused the error? Maybe the table name too

Reply:
Also can we make any column type a PK? Maybe support only the primitive types?

Reply:
I updated the error in [b85b61a](https://github.com/cloudquery/cloudquery/pull/20437/commits/b85b61a4adc235ecf574cf1ec964dc3ed27ba941)

Reply:
I would stay away from restricting the columns that can be used as a PK as this is an advanced feature... It all depends on how the destination handles a type. For example SQLite stringifies all complex columns so technically they could be used for a PK